### PR TITLE
Add integration test for the inclusive Gateway added

### PR DESCRIPTION
**Pull request by CTI777**
Related to https://github.com/Activiti/Activiti/issues/2739

### CODE REVIEW

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        ResponseEntity<CloudProcessInstance> processInstance = processInstanceRestTemplate.startProcess(
+                new StartProcessPayloadBuilder()
+                        .withProcessDefinitionKey(INCLUSIVE_GATEWAY_PROCESS)
+                        .withVariable("input",0)
+                        .build());
+        String processInstanceId = processInstance.getBody().getId();
+        String processDefinitionKey = processInstance.getBody().getProcessDefinitionKey();
+
+        //then
+        checkVariable(processInstanceId, "input", 0);
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java]: @CTI777 I think that there is no need to test variable creation here. This is already tested in other tests so we can trust the variable will be created.
Also, I think you can set the initial value to `1`, so you don't need to update the variable while completing the first task and we can keep the test as simple as possible (testing only events related to the inclusive gateway and the gateway behavior).

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+    public void testProcessExecutionWithInclusiveGateway() {
+        //when
+        streamHandler.getAllReceivedEvents().clear();
+        ResponseEntity<CloudProcessInstance> processInstance = processInstanceRestTemplate.startProcess(
+                new StartProcessPayloadBuilder()
+                        .withProcessDefinitionKey(INCLUSIVE_GATEWAY_PROCESS)
+                        .withVariable("input",0)
+                        .build());
+        String processInstanceId = processInstance.getBody().getId();
+        String processDefinitionKey = processInstance.getBody().getProcessDefinitionKey();
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java]: I don't think you need this variable: the process definition key will be the constant `INCLUSIVE_GATEWAY_PROCESS`

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+        
+        //when - complete first task
+        claimTask = taskRestTemplate.claim(task1);
+        assertThat(claimTask).isNotNull();
+        assertThat(claimTask.getBody().getStatus()).isEqualTo(TaskStatus.ASSIGNED);  
+ 
+        completeTaskPayload = TaskPayloadBuilder
+                                    .complete()
+                                    .withTaskId(task.getId())
+                                    .withVariables(Collections.singletonMap("input",3))
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java]: I don't think you need to update the variable value at this point as the execution has already passed the point where the variable is used for a decision.

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+        
+        //when - complete second task     
+        claimTask = taskRestTemplate.claim(task2);
+        assertThat(claimTask).isNotNull();
+        assertThat(claimTask.getBody().getStatus()).isEqualTo(TaskStatus.ASSIGNED);  
+ 
+        completeTaskPayload = TaskPayloadBuilder
+                                    .complete()
+                                    .withTaskId(task2.getId())
+                                    .withVariables(Collections.singletonMap("input",4))
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/InclusiveGatewayAuditProducerIT.java]: No need to update the variable here neither.

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Thanks @CTI777,

See my comments related to tests. 
Also, in this test you are covering the where the inclusive gateway is used to fork the path, I think we need to cover the case where it's used to merge paths as well.

**Comment by CTI777** [comment]
@erdemedeiros 
Refactoring for the test made

**Review by salaboy — APPROVED** [review]

**Review by salaboy — APPROVED** [review]

**Review by erdemedeiros — APPROVED** [review]

**Pull request merged**